### PR TITLE
Fix IsValidDefault for single-item list + add tests

### LINKED ISSUE #2334: Fix schema-first coercion of default values
Wait. What about coersion from a value to single-item list ?
```graphql
type Query {
  field1(arg: [String] = "aaa")
  field2(arg: [String] = ["aaa"])
}
```

field1 and field2 are effectively the same as I think.

_Originally posted by @sungam3r in https://github.com/graphql-dotnet/graphql-dotnet/pull/2321#discussion_r584118420_

### PR DESCRIPTION
**Pull request by sungam3r**
fixes #2334
also fixes behavior for non-null graph types in `ToAST`

### CODE REVIEW

**Comment by sungam3r** [comment]
@Shane32 I also added tests for `ToAST`. Look at my `TODO:` comments. What to do when calling `ToAST` with non-null graph type and `null` value? Exception?

**Review by Shane32 — APPROVED** [review]

File: src/GraphQL.Tests/Extensions/GraphQLExtensionsTests.cs [hunk 1]
```diff
@@ -4,20 +4,173 @@
...
+            public IEnumerator<object[]> GetEnumerator()
+            {
+                yield return new object[] { null, true, new NullValue() };
+
+                yield return new object[] { new BooleanGraphType(), true, new BooleanValue(true) };
+                yield return new object[] { new BooleanGraphType(), false, new BooleanValue(false) };
+                yield return new object[] { new BooleanGraphType(), null, new NullValue() };
+
+                yield return new object[] { new NonNullGraphType(new BooleanGraphType()), false, new BooleanValue(false) };
+                //yield return new object[] { new NonNullGraphType(new BooleanGraphType()), null, false }; //TODO: exception ?
```
Comment (sungam3r) [inline src/GraphQL.Tests/Extensions/GraphQLExtensionsTests.cs:97]: What about that case?

Comment (Shane32) [reply src/GraphQL.Tests/Extensions/GraphQLExtensionsTests.cs:97]: It seems reasonable to throw an `InvalidOperationException` if you are trying to coerce a value that is invalid for the type.  In this case, trying to cast a null value to a non-null type, so-to-speak.  Not much different than trying to convert the string "abc" to a BooleanGraphType AST representation.

**Comment by Shane32** [comment]
> @Shane32 I also added tests for `ToAST`. Look at my `TODO:` comments. What to do when calling `ToAST` with non-null graph type and `null` value? Exception? Something like that:
> 
> ```cs
>  public static IValue ToAST(this IGraphType type, object value)
>         {
>             if (type is NonNullGraphType nonnull)
>             {
>                 if (value == null)
>                     throw new InvalidArgumentException(nameof(value), "Unable to convert null value of the non-null type '{nonNull.ResolvedType}' to an AST representation.);
>                 return ToAST(nonnull.ResolvedType, value);
>             }
> ```
> 
> You can choose a better message.

Let's use `InvalidOperationException` to match the other exceptions in #2341 .  Or change them in both places.

**Comment by sungam3r** [comment]
OK, InvalidOperation.

**Comment by sungam3r** [comment]
@Shane32 I changed error messages a bit.

**Review by Shane32 — APPROVED** [review]

**Pull request merged**